runtime/src/comm/gasnet/comm-gasnet.c

```diff
@@ -118,6 +142,21 @@ typedef struct {
     done.flag = 0;                                                      \
   } while (0)
 
+static inline
+void do_wait_done(done_t* done)
+{
+#ifndef CHPL_COMM_YIELD_TASK_WHILE_POLLING
+  GASNET_BLOCKUNTIL(done->flag);
+#else
+  while (!done->flag) {
+    (void) gasnet_AMPoll();
+    chpl_task_yield();
+  }
+#endif
+}
+
+#define WAIT_DONE_OBJ(done) do_wait_done(&done)
```

Comment:
For safety, should be do_wait_done(&(done)), in case the macro arg has low-precedence operators in it. Alternatively, just have the users of this invoke do_wait_done(&whatever) themselves!